view/rooms/app.js

```diff
@@ -1009,6 +1011,50 @@ function setupColumns() {
   });
 }
 
+/**
+ * Infers roomType from url parameters other than roomType
+ * (used when it is not specified)
+ * @param {object} urlParams - JSON object of url params
+ * @returns {string} the first roomType name matching one of the url params
+ */
+function getRoomTypeFromMapping(urlParams) {
+  if (!_defaultRoomType) {
+    return '';
+  }
+  const urlParamNames = Object.keys(urlParams);
```

Comment:
any reason why not using forEach from here?

Reply:
you can't break out of a for each, so you'd have to loop through the entire array and then return the found value. This way we can exit the loop as soon as a match is found